From 006bc892fa132f4ba71ef23bb2043847bb3ece2e Mon Sep 17 00:00:00 2001
From: David Foley <favux.is@gmail.com>
Date: Mon, 27 Jun 2011 20:43:58 -0500
Subject: [PATCH] Add 3 new models: 6A, 6B, and 97

Adds the new Bamboo1 small (6A) and medium (6B).  Adds a slate form factor tablet PC (97).

Tested-by: Alex Tervoort <alex.tervoort@gmail.com> for 6A
Signed-off-by: David Foley <favux.is@gmail.com>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/wcmUSB.c            |    3 +++
 src/wcmValidateDevice.c |    1 +
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/wcmUSB.c b/src/wcmUSB.c
index 6ce2347..a5d5a61 100644
--- a/src/wcmUSB.c
+++ b/src/wcmUSB.c
@@ -242,6 +242,8 @@ static struct
 
 	{ WACOM_VENDOR_ID, 0x65, 100000, 100000, &usbBamboo     }, /* Bamboo */
 	{ WACOM_VENDOR_ID, 0x69,  39842,  39842, &usbBamboo1    }, /* Bamboo1 */
+	{ WACOM_VENDOR_ID, 0x6A, 100000, 100000, &usbBamboo1    }, /* Bamboo1 4x6 */
+	{ WACOM_VENDOR_ID, 0x6B, 100000, 100000, &usbBamboo1    }, /* Bamboo1 5x8 */
 
 	{ WACOM_VENDOR_ID, 0xB0, 200000, 200000, &usbIntuos3    }, /* Intuos3 4x5 */
 	{ WACOM_VENDOR_ID, 0xB1, 200000, 200000, &usbIntuos3    }, /* Intuos3 6x8 */
@@ -265,6 +267,7 @@ static struct
 
 	{ WACOM_VENDOR_ID, 0x90, 100000, 100000, &usbTabletPC   }, /* TabletPC 0x90 */
 	{ WACOM_VENDOR_ID, 0x93, 100000, 100000, &usbTabletPC   }, /* TabletPC 0x93 */
+	{ WACOM_VENDOR_ID, 0x97, 100000, 100000, &usbTabletPC   }, /* TabletPC 0x97 */
 	{ WACOM_VENDOR_ID, 0x9A, 100000, 100000, &usbTabletPC   }, /* TabletPC 0x9A */
 	{ WACOM_VENDOR_ID, 0x9F, 100000, 100000, &usbTabletPC   }, /* CapPlus  0x9F */
 	{ WACOM_VENDOR_ID, 0xE2, 100000, 100000, &usbTabletPC   }, /* TabletPC 0xE2 */
diff --git a/src/wcmValidateDevice.c b/src/wcmValidateDevice.c
index 456a113..2a6c0bf 100644
--- a/src/wcmValidateDevice.c
+++ b/src/wcmValidateDevice.c
@@ -261,6 +261,7 @@ int wcmDeviceTypeKeys(InputInfoPtr pInfo)
 		case 0x93: /* TPC with 1FGT */
 		case 0x9A: /* TPC with 1FGT */
 		case 0x90: /* TPC */
+		case 0x97: /* TPC */
 			TabletSetFeature(priv->common, WCM_TPC);
 			break;
 
-- 
1.7.0.1

